src/orchard/util/io.clj

```diff
@@ -22,3 +27,52 @@
             (System/setOut old-out))
           (when (= ps System/err) ;; `System/err` may have changed in the meantime (in face of concurrency)
             (System/setErr old-err)))))))
+
+(defn url-protocol
+  "Get the URL protocol as a string, e.g. http, file, jar."
+  [^java.net.URL url]
+  (.getProtocol url))
+
+(defn in-archive?
+  "Does this URL point to a file inside a jar (or zip) archive.
+  i.e. does it use the jar: protocol."
+  [^java.net.URL url]
+  (= "jar" (url-protocol url)))
+
+(defn jar-location
```

Comment:
`jar-location-of-file-within-archive`?

More verbose, but:

* consistent w/ my prior suggestion
* does not take a generic name for future use cases

Reply:
I'm fine with a shorter name as the longer seems extremely verbose. I'm all for clear naming, but we don't want to go overboard here.

Reply:
this is now `resource-jarfile`, since it returns the `.jar` `File` instance for a given resource (resources being URLs).